src/custom/components/Header/NetworkSelector/NetworkSelectorMod.tsx

```diff
@@ -379,7 +379,7 @@ export default function NetworkSelector() {
   )
 
   useEffect(() => {
-    if (!chainId || !previousChainId) return
+    if (!chainId) return
```

Comment:
This is the actual fix.

The network change was only respected after the initial ID was set.

The code is almost identical to original uniswap's so I can't really tell what broke.
Maybe there was a bug on their side already fixed 🤷

Reply:
this is actually broken on uniswap interface too so if this works i'll probably add this to the uniswap interface. or feel free to make a pr.

Reply:
The fix is changing slightly the behaviour.

Now when connected, the wallet chainId always takes precedence.
Changing the URL won't trigger a network change in the wallet to avoid race conditions